### PR TITLE
Add local aliases to default host cert principals

### LINKED ISSUE #2910: "localhost" not in the set of valid principals for given certificate
**What happened**:
When a customer completes the Quickstart they get this error message. 

```
WARNING: You are using insecure connection to SSH proxy 
https://localhost:3080 error: ssh: handshake failed: ssh: principal 
"localhost" not in the set of valid principals for given certificate: 
["cc91fdc6-81a8-4dc2-88d0-77b4b07e1b16.abdu-ThinkPad-T480" "abdu-ThinkPad-
T480.abdu-ThinkPad-T480" "abdu-ThinkPad-T480" 
"remote.kube.proxy.teleport.cluster.local"]
```

This is due to `localhost` not being created as part of the SSH certificate. Customers can fix this by copying the hostname and using this e.g. `https://abdu-ThinkPad-T480:3080`.

**What you expected to happen**:
We should improve the error message, if possible add the hostname as default. 

Customer can add localhost to the set of principles by adding 
```
teleport:
  proxy_service:
    listen_addr: localhost:3080
```

**How to reproduce it (as minimally and precisely as possible)**:

**Environment**:
- Teleport version (use `teleport version`): 4.0
- Tsh version (use `tsh version`):
- OS (e.g. from /etc/os-release): Linux

### PR DESCRIPTION
**Pull request by awly**
Adding following principals:
- `localhost`
- `127.0.0.1`
- `::1`

With these, `tsh` (both `ssh` and `join`) works with a local proxy
without any SSH handshake errors.

Removed the warning from quickstart docs, but keeping `--proxy=grav-00`
since that implies to the reader that proxy is usually remote.

Fixes #2910

### CODE REVIEW

File: lib/auth/native/native_test.go
Comment (russjones) [inline lib/auth/native/native_test.go:166]: We try and make log lines full sentences, so:

```go
c.Logf("Running test case: %q.", tt.desc)
```

File: lib/auth/native/native.go
Comment (russjones) [inline lib/auth/native/native.go:317]: We have a constants file: https://github.com/gravitational/teleport/blob/master/constants.go

**Review by russjones — APPROVED** [review]

File: lib/auth/native/native.go
Comment (awly) [reply lib/auth/native/native.go:317]: Moved.
Also switched to a typed constant. Would be nice to use those more instead of opaque strings everywhere (for readability and type safety reasons)

File: lib/auth/native/native_test.go
Comment (awly) [reply lib/auth/native/native_test.go]: Done

**Comment by awly** [comment]
Ping @benarent or @webvictim for CODEOWNERS approval

**Review by benarent — APPROVED** [review]
nice! this makes the quickstart much smoother.

**Comment by benarent** [comment]
@awly Feel free to delete branches after merging.

**Pull request merged**
